lib/components/rails/attributes.rb

```diff
@@ -3,8 +3,20 @@
 module Components
   module Rails
     module Attributes
+      extend ActiveSupport::Concern
+
       def _render_template(options)
-        super(options.merge(locals: attributes))
+        super(options.merge(locals: self.class.default_attributes.merge(attributes)))
+      end
+
+      module ClassMethods
+        def defaults(default_attributes)
```

Comment:
@joeosburn-desiringgod why not just call this method `default_attributes`? Because it collides with the one below?

Reply:
@benhutton We should. That's up for debate. I prefer the simplicity of `defaults`, and then if we ever have other things that need defaults, we could deal with the naming conflicts then. I'm guessing that'll never happen.

Reply:
@joeosburn-desiringgod I guess my problem isn't with what the public API is `defaults` vs `default_attributes`, but more with having two names in play. I agree that we likely won't have other things that need defaults, and so it's best to just assume we won't for now. In that case, why not just get rid of `_attributes` everywhere here?

```
  def defaults(defaults = nil)
    if defaults.present?
      @defaults = defaults
    else
      @defaults ||= {}
    end
  end
```

Reply:
@benhutton That's good. I will make that change. Thanks.

Reply:
@benhutton Actually, I don't like this.

Reply:
I'm trying to think of a comparative method that does something similar. When do we have the same method that's the getter and the setter. I can't think of anything. Can you?